### PR TITLE
[jvmfinder] add registry HKLM/JavaSoft/JRE to searched places

### LINKED ISSUE #345: Windows: JVMFinder.get_jvm_path() cannot find jvm.dll via registry when JRE is installed as a part of Oracle JDK
JVMFinder.get_jvm_path() cannot find jvm.dll because it tries the JAVA_HOME sys var first and when it's not found, checks the registry path branch:
'HKEY_LOCAL_MACHINE\SOFTWARE\JavaSoft\Java Runtime Environment'
this branch is created only by Oracle JRE installer.

When user installs JDK with JRE included, the branch has a different name ('JRE' instead of 'Java Runtime Environment').

exception raised on jaydebeapi.connect(<my args>) call
: Exception: No JVM shared library file (jvm.dll) found. Try setting up the JAVA_HOME environment variable properly
lib\site-packages\jpype\_jvmfinder.py:153: JVMNotFoundException

Workaround: rename it manually.

JPype1 version: 0.6.3
JDK version: jdk-10.0.2_windows-x64_bin.exe
Windows 10

### PR DESCRIPTION
**Pull request by marscher**
Fixes #345

### CODE REVIEW

**Comment by Thrameos** [comment]
Hmm.  Did a problematic patch go in?  Looks like something needs to be resolved with the CI.

**Comment by marscher** [comment]
the usage of the preinstalled jdk speeds up Appveyor significantly. @Thrameos was there any reason to switch to the chocolatey version in the first place?

**Comment by Thrameos** [comment]
I do not recall why I made a switch to using chocolaty.  I think it was a matter of convenience as I believe we were manually installing pieces we needed at some point. 

Any system the works to keep th CI going is fine with me.

File: appveyor.yml [hunk 1]
```diff
@@ -72,11 +73,12 @@ install:
   # - We will need to figure out how to run nose multiple times on the same run so that
   #   we can get the result from each of the java version runs.
   #
-  - cinst jdk8 -params 'installdir=C:\\jdk8' %CINST_OPTS%
+  #- cinst jdk8 -params 'installdir=C:\\jdk8' %CINST_OPTS%
 
   # - cinst jdk7 -params 'installdir=C:\\jdk7'
   # - cinst jdk9 -version 9.0.4.11 -params 'installdir=C:\\jdk9'
-  - cinst ant %CINST_OPTS%
+  - cinst ant %CINST_OPTS% --ignore-dependencies
```
[the remark this answers is not in the document]
Comment (marscher) [reply appveyor.yml]: I tried it, but the passed value was ignored. So it might follow another install location parameter...

**Comment by Thrameos** [comment]
I think this whole patch should get pulled into devel as well.   Everything looks good so it is just a matter of cleaning up any conflicts.  I can try and give a shot in a day or two.

**Comment by marscher** [comment]
I can merge/push it to devel.

**Comment by Thrameos** [comment]
That would be great.  Thanks!

**Pull request merged**